### PR TITLE
feat: Add _mm_aesimc_si128

### LINKED ISSUE #566: `_mm_aesimc_si128` is absent
[_mm_aesimc_si128](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_aesimc_si128) is absent. It perform the [AES InvMixColumns transformation](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.197.pdf). After implementing this intrinsics, SSE2NEON should reach 100% coverage.

### PR DESCRIPTION
**Pull request by howjmay**
closes #566

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -9746,6 +9747,57 @@ FORCE_INLINE __m128i _mm_aesdeclast_si128(__m128i a, __m128i RoundKey)
...
+    uint8x16_t v = vreinterpretq_u8_m128i(a);
+    uint8x16_t w;
+
+    // muliplying 'v' by 4 in GF(2^8)
+    w = (v << 1) ^ (uint8x16_t) (((int8x16_t) v >> 7) & 0x1b);
+    w = (w << 1) ^ (uint8x16_t) (((int8x16_t) w >> 7) & 0x1b);
+    v ^= w;
+    v ^= (uint8x16_t) vrev32q_u16((uint16x8_t) w);
+
+    w = (v << 1) ^ (uint8x16_t) (((int8x16_t) v >> 7) &
```
Comment (jserv) [inline sse2neon.h:9766]: Move the comment "muliplying 'v' by 2 in GF(2^8)" before the statement `w = `.

Comment (jserv) [reply sse2neon.h:9766]: In addition, fix typo in comment.

File: sse2neon.h [hunk 2]
```diff
@@ -9623,18 +9623,19 @@ FORCE_INLINE __m128i _mm_aesdec_si128(__m128i a, __m128i RoundKey)
...
-     ((y >> 3 & 1) * XT(XT(XT(x)))) ^ ((y >> 4 & 1) * XT(XT(XT(XT(x))))))
-
+    /* FIXME: optimized for NEON */
     uint8_t i, e, f, g, h, v[4][4];
     uint8_t *_a = (uint8_t *) &a;
     for (i = 0; i < 16; ++i) {
         v[((i / 4) + (i % 4)) % 4][i % 4] = SSE2NEON_rsbox[_a[i]];
     }
 
+#define XT(x) (((x) << 1) ^ ((((x) >> 7) & 1) * 0x1b))
```
Comment (jserv) [inline sse2neon.h:9633]: There are potentially conflicts when you define macro `XT` and `MULTIPLY`. You have to add `SSE2NEON_` prefix.

Comment (jserv) [reply sse2neon.h:9633]: By the way, you should modify the existing usage of `XT` and `MULTIPLY` as well. I forgot to raise the concerns.

File: sse2neon.h [hunk 3]
```diff
@@ -9746,6 +9749,59 @@ FORCE_INLINE __m128i _mm_aesdeclast_si128(__m128i a, __m128i RoundKey)
 #endif
 }
 
+// Perform the InvMixColumns transformation on a and store the result in dst.
+// https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_aesimc_si128
+FORCE_INLINE __m128i _mm_aesimc_si128(__m128i a)
+{
+#if defined(__aarch64__)
+    static const uint8_t ror32by8[] = {0x1, 0x2, 0x3, 0x0, 0x5, 0x6, 0x7, 0x4,
+                                       0x9, 0xa, 0xb, 0x8, 0xd, 0xe, 0xf, 0xc};
```
Comment (jserv) [inline sse2neon.h:9758]: Add a comma before `}`.

Comment (howjmay) [reply sse2neon.h:9758]: done

**Comment by howjmay** [comment]
I have added another commit that makes XT and MULTIPLY as shared macros

**Pull request merged**